report_templates/ansible_inventory.erb

```diff
@@ -1,144 +0,0 @@
-<%#
-name: Ansible Inventory
-description: This template has a special meaning and should not be renamed.
-snippet: false
-template_inputs:
-- name: Organization
-  required: true
-  input_type: user
-  description: Whether should be Organization included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: Location
-  required: true
-  input_type: user
-  description: Whether should be Location included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: Host Group
-  required: true
-  input_type: user
-  description: Whether should be Host Group included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: Host Collections
-  required: true
-  input_type: user
-  description: Whether should be Host Collections included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: IPv4
-  required: true
-  input_type: user
-  description: Whether should be IPv4 included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: IPv6
-  required: true
-  input_type: user
-  description: Whether should be IPv6 included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: no
-- name: Subnet
-  required: true
-  input_type: user
-  description: Whether should be Subnet included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: Smart Proxies
-  required: true
-  input_type: user
-  description: Whether should be Smart Proxies included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: yes
-- name: Content Attributes
-  required: true
-  input_type: user
-  description: Whether should be Content Attributes included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: no
-- name: Facts
-  required: true
-  input_type: user
-  description: Whether should be Facts included.
-  advanced: false
-  value_type: plain
-  options: "yes\r\nno"
-  default: no
-model: ReportTemplate
-require:
-- plugin: foreman_ansible
-  version: '>= 3.1.0'
-- plugin: katello
```

Comment:
Can it be made so that Katello is optional and it detects whether Katello is installed?

Reply:
There's no good mechanism for that, we have macro to check if plugin is present but input "include content attributes" would still be available and silently ignored. Given the inventory template is configurable through settings, I'd start with this and people could easily create alternative, dropping some attributes.

Reply:
That's fair. In a way this is becoming an API so it'd need a way to raise an error when an invalid combination was requested.